test/e2e/specs/editor/various/autocomplete-and-mentions.spec.js

```diff
@@ -426,13 +460,20 @@ test.describe( 'Autocomplete (@firefox, @webkit)', () => {
 
 	test( 'should hide UI when selection changes (by mouse)', async ( {
 		page,
+		editor,
+		pageUtils,
 	} ) => {
-		await page.click( 'role=button[name="Add default block"i]' );
-		await page.keyboard.type( '@fr' );
+		await editor.canvas.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '@' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'b' );
+		await page.keyboard.type( 'f' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'b' );
+		await page.keyboard.type( 'r' );
 		await expect(
 			page.locator( 'role=option', { hasText: 'Frodo Baggins' } )
 		).toBeVisible();
-		await page.click( '[data-type="core/paragraph"]' );
+		// Use the strong tag to move the selection by mouse within the mention.
+		await editor.canvas.click( '[data-type="core/paragraph"] strong' );
```

Comment:
This test was actually passing for the wrong reason. The autocomplete component has an on click outside handler that hides the popover, which doesn't work in the iframe when clicking in a block.

In the test we should perform a click that changes the selection but still within the mention.